labellab-flask/api/controllers/imagescontroller.py

```diff
@@ -78,7 +77,7 @@ def post(self, project_id):
             images_new = []
             for i in range(len(images)):
                 image = images[i]
-                image_name = image.filename.split('.')[0]
+                image_name = image_name if len(images) == 1 else image.filename.split('.')[0]
```

Comment:
```suggestion
                image_name = image_names[i] or image.filename.split('.')[0]
```

Reply:
For this to work, ensure we're passing `image_names` array of same length.
If the user is not providing custom file name, pass undefined in the array.
e.g. `[undefined]`

Or you can pass the default file name in the `image_names` array and skip the `or` condition altogether.

Reply:
All changes done!